### PR TITLE
Pimp out your Hardsuits

### LINKED ISSUE #35838: Feature Request: Pimp out your Hardsuits
Requesting a better way to paint hardsuits that isn't just slapping a color matrix over the whole thing but rather recoloring it piece by piece. I have done the legwork dividing the engineering hardsuit sprites into what I think are the key zones for recoloring:
- The visors and sensors
- Primary Shell
- Inner Shell

I don't know how to code this myself hence this feature request.
if it shows promise then I'm fully on board with doing the sprite separation for the other hardsuits.

I think the mechpainter could be a good candidate for painting hardsuits (and could possibly rename it to exoshell-painter or something).

[spriteworkdone]
[featurerequest]

image

Link to the above DMI:
https://cdn.discordapp.com/attachments/1084300577862856857/1198854429944860742/engi-OVerlays.dmi?ex=65c06b31&is=65adf631&hm=ba45d050f7671b20c5fe1a324e4a20179cde8c6c70392924d462b39a6bf9c1d0&

### PR DESCRIPTION
**Pull request by DeityLink**
Fixes #35838

image

Sprites provided by @RealestEstate , work in progress.

:cl:
* rscadd: Engineering hardsuits can have their colours customized by using a painting brush or roller on them with some paint. Deploy the helmet to paint it as well.

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -296,6 +310,7 @@
 	if (hood && (mannequin.clothing[SLOT_MANNEQUIN_HEAD] == hood))
 		hooddown(mannequin)
 
+//for fabric clothing can can be fully dyed in a cauldron
```
Comment (Securitymain) [inline code/modules/clothing/clothing.dm]: do the can-can

**Comment by Blithering** [comment]
If you're gonna be expanding this to include secondary colours on a suit, could that be rolled into the suit modifier machine?

**Comment by DeityLink** [comment]
I had never used the exosuit modification machine, spawned one to try it, found it slow and cumbersome and decided that allowing people to directly use a painting brush was the simplest option.

Also there ARE secondary colors here, it's just that the secondary parts are way too dark and show almost the same colour on all 3 suits. And the visors/hand light are a third layer.

**Comment by RealestEstate** [comment]
I'll work on this some more in the coming week.

**Comment by DeityLink** [comment]
same, I just remembered yesterday to merge your PR.

**Comment by ConfusedCactus** [comment]
is there a paint color that would make this look like a nukie hardsuit

**Comment by DeityLink** [comment]
uh, kinda @ConfusedCactus 
image

**Comment by RealestEstate** [comment]
I sent a PR your way with the new greyscale for everything.
these are some examples
PROOF OF CONCEPT

PR needs a way to clean paint off the hardsuits, and after that I think it should be ready for review.

**Comment by DeityLink** [comment]
Thanks, I'll take care of this first after the upcoming cult update.

**Comment by DeityLink** [comment]
so like, very soon now

**Comment by jwhitak** [comment]
Here's to seeing some of the creative hardsuits in rounds to come...

**Pull request merged**